Improvement/mk8 s 68 use smart cron trigger

rdebay-scality opened the pull request:
**Component**: crons for nightly and builds

**Context**:  Have more efficient nightlies. Only run when needed

**Summary**:
* Use smart triggers
* builds schedule to avoid losing the needed artefacts for nightlies
* Auto retry with slack messages

Closes: MK8S-68

rdebay-scality committed:
MK8S-68: Use smart trigger for crons

bert-e commented:
# Hello rdebay-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/metalk8s/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

rdebay-scality committed:
MK8S-68: Make sure we have a build

Since Artefacts are removed every two weeks, make sure
we have a build. At least it is started once a week.

rdebay-scality commented:
/create_integration_branches

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/MK8S-68-use-smart-cron-trigger` into
`development/132.0`
* [w/133.0/improvement/MK8S-68-use-smart-cron-trigger](https://github.com/scality/metalk8s/compare/development/133.0...w/133.0/improvement/MK8S-68-use-smart-cron-trigger) will be merged into `development/133.0` 



The following branches will **NOT** be impacted:

* `development/123.0`
* `development/124.0`
* `development/124.1`
* `development/125.0`
* `development/126.0`
* `development/127.0`
* `development/128.0`
* `development/129.0`
* `development/130.0`
* `development/131.0`
* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbwatenbergscality

* @teddyandrieux

* @eg-ayoub

* @thomasdanan

* @chengyanjin

* @ezekiel-alexrod

* @m4nch0t

* @anthony-treuillier-scality

* @xaf-scality

* @rdebay-scality

* @vdaviot










*The following options are set:* **create_integration_branches**

ezekiel-alexrod commented on .github/workflows/crons.yaml:
```diff
@@ -33,8 +33,10 @@ jobs:
         if: github.event.schedule == matrix.cron
         uses: actions/checkout@v6
-      - name: Run ${{ matrix.name }}
+      - name: 'Smart trigger for ${{ matrix.name }}'
```
nit: When possible, prefer double quotes

ezekiel-alexrod commented on .github/workflows/crons.yaml:
```diff
@@ -4,6 +4,9 @@ on:
   # NOTE: Schedule is only run on the latest commit of the default branch,
   # so we need to schedule Cron Jobs for all branches here
   schedule:
+    # Daily builds, scheduled before nightlys every weekday at 22:00
```
typo: nightlies

bert-e commented:
# History mismatch



Merge commit #a47b85cc30a85ae949f77754b70b3ff0ce6af466 on the integration branch
`w/133.0/improvement/MK8S-68-use-smart-cron-trigger` is merging a branch which is neither the current
branch `improvement/MK8S-68-use-smart-cron-trigger` nor the development branch
`development/133.0`.

It is likely due to a rebase of the branch `improvement/MK8S-68-use-smart-cron-trigger` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

rdebay-scality committed:
MK8S-68: add retry mechanism for failed nightlies with Slack notifications

Add automatic retry of failed nightly workflows:
- Retry schedule at 06:00 (4 hours after last nightly)
- Maximum 2 retry attempts for failed jobs only
- Slack notifications to #squad-supsetup-alerts when:
  * A retry is triggered
  * Manual action is needed after max retries exhausted
- Uses scality/actions/action-retry-workflow@1.17.0

rdebay-scality commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_integration_branches**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/MK8S-68-use-smart-cron-trigger` into
`development/132.0`
* [w/133.0/improvement/MK8S-68-use-smart-cron-trigger](https://github.com/scality/metalk8s/compare/development/133.0...w/133.0/improvement/MK8S-68-use-smart-cron-trigger) will be merged into `development/133.0` 



The following branches will **NOT** be impacted:

* `development/123.0`
* `development/124.0`
* `development/124.1`
* `development/125.0`
* `development/126.0`
* `development/127.0`
* `development/128.0`
* `development/129.0`
* `development/130.0`
* `development/131.0`
* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @eg-ayoub

* @rdebay-scality

* @ezekiel-alexrod

* @thomasdanan

* @m4nch0t

* @anthony-treuillier-scality

* @vdaviot

* @jbwatenbergscality

* @chengyanjin

* @xaf-scality

* @teddyandrieux










*The following options are set:* **create_integration_branches**

rdebay-scality commented:
/approve

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/132.0`

* :heavy_check_mark: `development/133.0`



The following branches have **NOT** changed:

* `development/123.0`
* `development/124.0`
* `development/124.1`
* `development/125.0`
* `development/126.0`
* `development/127.0`
* `development/128.0`
* `development/129.0`
* `development/130.0`
* `development/131.0`
* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue MK8S-68.

Goodbye rdebay-scality.




*The following options are set:* **approve, create_integration_branches**
